external/mnist_data.cmake

```diff
@@ -0,0 +1,50 @@
+# -----------------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# -----------------------------------------------------------------------------
+
+# Fetch MNIST dataset from online archive
+#
+if(EXISTS ${REPOSITORY_DIR}/build/ThirdParty/share/eigen-master.zip)
+    set(URL ${REPOSITORY_DIR}/build/ThirdParty/share/eigen-master.zip)
+else()
+    set(URL "https://github.com/wichtounet/mnist/archive/master.zip")
+    set(HASH "0bcecef3aaed7c619f0baac02697a76dff3da54f0c17f1aaaa92a2928ca300f3")
+endif()
+
+message(STATUS "obtaining MNIST data")
+include(DownloadProject/DownloadProject.cmake)
```

Comment:
why do we use DownloadProject, instead of ExternalProject, which seems more common and part of cmake?

Reply:
DownloadProject is a workaround because the CMake's equivalent feature is too recent of a version.  Anyway, DownloadProject does use ExternalProject to do the job.

Reply:
> CMake's equivalent feature is too recen 

I recall, it was CMake even not in recent Ubuntu. Thanks